container-search/src/main/java/com/yahoo/search/dispatch/Dispatcher.java

```diff
@@ -222,12 +224,28 @@ private void processResponse(Client.GetDocsumsResponseOrError responseOrError) {
         }
 
         private void fill(List<FastHit> hits, byte[] slimeBytes) {
-            Inspector summaries = new SlimeAdapter(BinaryFormat.decode(slimeBytes).get().field("docsums"));
+            com.yahoo.slime.Inspector root = BinaryFormat.decode(slimeBytes).get();
+            Inspector summaries = new SlimeAdapter(root.field("docsums"));
             if ( ! summaries.valid())
                 throw new IllegalArgumentException("Expected a Slime root object containing a 'docsums' field");
             for (int i = 0; i < hits.size(); i++) {
                 fill(hits.get(i), summaries.entry(i).field("docsum"));
             }
+            com.yahoo.slime.Inspector errors = root.field("errors");
+            errors.traverse((ArrayTraverser)(int index, com.yahoo.slime.Inspector value) -> {
+                ByteArrayOutputStream os = new ByteArrayOutputStream(1024);
+                try {
+                    new JsonFormat(true).encode(os, value);
```

Comment:
How about we define the content of this instead of passing JSON through?
I suggest a mandatory "type" and "message" field and an optional "details" field, as that can be mapped cleanly to ErrorMessage.

Reply:
Done.